lib/ansible/modules/cloud/azure/azure_rm_routetable.py

```diff
@@ -0,0 +1,199 @@
+#!/usr/bin/python
+#
+# Copyright (c) 2018 Yuwei Zhou, <yuwzho@microsoft.com>
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: azure_rm_routetable
+version_added: "2.7"
+short_description: Manage Azure route resource.
+description:
+    - Create, update or delete a route.
+options:
+    resource_group:
+        description:
+            - name of resource group.
+        required: true
+    name:
+        description:
+            - name of the virtual network.
+        required: true
+    state:
+        description:
+            - Assert the state of the virtual network. Use 'present' to create or update and
+              'absent' to delete.
+        default: present
+        choices:
+            - absent
+            - present
+    disable_bgp_route_propagation:
+        description:
+            - Gets or sets whether to disable the routes learned by BGP on that route table.
+            - True means disable.
+        type: bool
+        default: False
+    location:
+        description:
+            - Region of the resource.
+            - Derived from C(resource_group) if not specified
+
+extends_documentation_fragment:
+    - azure
+    - azure_tags
+
+author:
+    - "Yuwei Zhou (@yuwzho)"
+
+'''
+
+EXAMPLES = '''
+    - name: Create a route table
```

Comment:
We should probably also include a sample and/or doc note that shows that the route table is associated via `azure_rm_subnet` - was definitely the simpler choice to implement that way, but we need to tell people where that happens.

Reply:
From SDK interface, we cannot associate the subnet from route table, but give route table id to subnet. I think it needs another PR to implement.